### ISSUE TITLE
Get-PnPSiteTemplate mixes up pages and retrieves old content

### ISSUE THREAD

**Issue by lennertcc** [pnp/pnpframework#204]
I am using the following command:

`Get-PnPSiteTemplate -Out Export.xml -Handlers Pages,PageContents -IncludeAllClientSidePages`

The result is completely mixed up. I get a <pnp:ClientSidePage> with a Title and a PageName combination that does not exist in my library, but both Title and PageName do exist on separate pages in the same library.

Also I am getting old content. Today I added a few webparts, updated the content of a Text webpart and published a major version. These changes are not reflected in the result. Also I am getting a two column section in a particular ClientSidePage that does not exists on the page in SharePoint.

Earlier this week there was no problem with the same commandlet. Today I upgraded my PowerShell to:
Manifest   1.3.18     nightly    PnP.PowerShell                    

It seems that this also caused an upgrade to 
`<pnp:Preferences Generator="PnP.Framework, Version=1.2.22.0, Culture=neutral, PublicKeyToken=null" />`
coming from 1.2.21.0 before the upgrade.

Using SharePointPnPPowerShellOnline (Binary     3.29.21... SharePointPnPPowerShellOnline) 
`Get-PnPProvisioningTemplate -Out Export.xml -Handlers Pages,PageContents -IncludeAllClientSidePages`

is giving me the expected results.

**Comment by jansenbe** [comment]
@lennertcc : we had some issues with a wrong site pages library being detected (e.g. if you also had a wiki page library in the same site)...can you still reproduce this with the latest version?

**Comment by lennertcc** [comment]
Hi Bert, thank you for working on this. I see a small difference in the output, but still it's completely different from what I get using the old Get-PnPProvisioningTemplate. I have been working with PnP templating for a long time now and I find it hard to understand how it retrieves old content. I double checked all versions of the used modules and double checked I am exporting the same site collection each time.

I have exports of the different Module versions. I will ask the owner of the site for permission to share it. Otherwise I can maybe share it privately or in a screensharing session.

**Comment by jansenbe** [comment]
@lennertcc : Are you using PS7 or PS5?

**Comment by lennertcc** [comment]
PS7.

I did not get permission to share the templates as it contains too much information about the company and the people working there. I can however have a screen sharing session to show the behavior. Just contact me if that's helpfull.

**Comment by jansenbe** [comment]
@lennertcc : I would need to have a repro, if this cannot be simulated outside of the customer's context it's hard to fix.

One thing to however try (if possible for the customer) is to install PS 5 with the latest PnP PowerShell module and then try the same...we've seen that the XML serialize in .NET Core (the one used by PS7) sometimes changes the order of elements...maybe this is related. Trying with PS5 means you're using the .NET Framework version of PnP PowerShell which does not exhibit this problem. Thanks for trying.

**Comment by lennertcc** [comment]
I have tested it with PowerShell 5.1.19041.610. Still the same differences. Let me highlight one example that does not contain any contextual information.

PnP.Powershell produces:
```
<pnp:Provisioning xmlns:pnp="http://schemas.dev.office.com/PnP/2020/02/ProvisioningSchema">
  <pnp:Preferences Generator="PnP.Framework, Version=1.3.0.0, Culture=neutral, PublicKeyToken=null" />
[...]
        <pnp:ClientSidePage PromoteAsNewsArticle="false" PromoteAsTemplate="false" Overwrite="true" EnableComments="false" Title="Page X" ContentTypeID="0x0101009D1CB255DA76424F860D91F20E6C411800C39EC0D9E347CE4FA1015BEFEB1E04CF" ThumbnailUrl="" PageName="Search.aspx">
          <pnp:Header Type="Default" LayoutType="NoImage" TextAlignment="Center" ShowTopicHeader="false" ShowPublishDate="false" TopicHeader="" AlternativeText="" Authors="[]" AuthorByLine="[]" AuthorByLineId="-1" />
          <pnp:Sections>
            <pnp:Section Order="1" Type="TwoColumn">
[...]
```

SharePointPnpPowerShellOnline produces:
```
<pnp:Provisioning xmlns:pnp="http://schemas.dev.office.com/PnP/2020/02/ProvisioningSchema">
  <pnp:Preferences Generator="OfficeDevPnP.Core, Version=3.28.2012.0, Culture=neutral, PublicKeyToken=5e633289e95c321a" />
[...]
        <pnp:ClientSidePage PromoteAsNewsArticle="false" PromoteAsTemplate="false" Overwrite="true" EnableComments="false" Title="Page X" ContentTypeID="0x0101009D1CB255DA76424F860D91F20E6C411800C39EC0D9E347CE4FA1015BEFEB1E04CF" ThumbnailUrl="" PageName="Search.aspx">
          <pnp:Header Type="Default" LayoutType="FullWidthImage" TextAlignment="Center" ShowTopicHeader="false" ShowPublishDate="false" TopicHeader="" AlternativeText="" Authors="[]" AuthorByLine="[]" AuthorByLineId="-1" />
          <pnp:Sections>
            <pnp:Section Order="1" Type="OneColumn" BackgroundEmphasis="Neutral">
[...]
```

Now this page Search.aspx does not contain any TwoColumn sections. It also has a FullWitdthImage with the Title ('Page X') instead of NoImage. I have tried to make sense of the difference, but I cannot find any good explanation. 

From other differences in the result I still get the feeling that some old version of some of the pages is retrieved by PnP.PowerShell. These pages are frequently changing and I also see guids of custom webparts we used to have on those pages. Could it be that there is some caching mechanism somewhere?

**Comment by jansenbe** [comment]
@lennertcc : are there draft versions of these pages? Might be that the draft page is picked up? Do you see the same behavior with published pages?

**Comment by lennertcc** [comment]
Now I have an idea where this comes from. We have been renaming the pages that I see the problems with. So this happened:

- We started out with Search-Hub.aspx
- We renamed or copied this page to have it as Search.aspx
- Still in the same library we also have Search-Hub.aspx
- In the output xml when using PnP.Powershell I now see <pnp:ClientSidePage [...] PageName="Search.aspx"> but the content there is from the page Search-Hub.aspx
- In the same output I also have <pnp:ClientSidePage [...] PageName="Search-Hub.aspx"> with the exact same content as <pnp:ClientSidePage [...] PageName="Search.aspx">
- The actual content of Search.aspx it not in the export.

I'm pretty sure this is the way to reproduce this bug as I have the same problem with two pages where we did the exact same thing (the other being News-Hub.aspx to News.aspx).
